### PR TITLE
Add monitor mode

### PR DESCRIPTION
**Pull request by viktorasm**
Adding alternative agent process execution mode: monitor. This is a lightweight agent mode meant to monitor itself, and deliver agent's own execution events to CAST AI. 

Info between agent and monitor will be exchanged by local "metadata" file - we need to communicate clusterID to monitor so that it's able to send logs on it's own, plus we ship "main agent process ID" so that monitor can detect restarts.

Restarts are be detected by main agent process ID change. Once agent restarts, monitor will detect a oom-related crash and report it to CAST AI via logs.

Two types of events will be sent to CAST AI:
* agent pod related warnings (filtered by namespace and pod name)
* agent pod node warnings, filtered by "castai-agent" constant (node events do not have any metadata relating event to the pod)

some additional boyscouting like makefile updates, and bumping golang 1.19 -> 1.20.4

A complimentary helm update: https://github.com/castai/helm-charts/pull/264/files

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,16 +1,11 @@
 build:
 	GOOS=linux GOARCH=amd64 CGO_ENABLED=0 go build -ldflags "-s -w" -o bin/castai-agent-amd64 .
-	docker build -t us-docker.pkg.dev/castai-hub/library/agent:$(VERSION) .
+	GOOS=linux GOARCH=arm64 CGO_ENABLED=0 go build -ldflags "-s -w" -o bin/castai-agent-arm64 .
+	docker buildx build --push --platform=linux/amd64,linux/arm64 -t  $(TAG) .
```
Comment (viktorasm) [inline Makefile:4]: boyscout: fixing this make target after changes were done for multi-arch build.

File: Makefile [hunk 2]
```diff
@@ -1,16 +1,11 @@
...
 
 generate:
 	go generate ./...
 
-push:
```
Comment (viktorasm) [inline Makefile:8]: deleting some unused make targets. for local development, "build" now both builds and pushes multi-arch image.

File: main.go [hunk 3]
```diff
@@ -161,6 +172,27 @@ func run(ctx context.Context, castaiclient castai.Client, log *logrus.Entry, cfg
 			log.Infof("clusterID: %s provided by env variable", clusterID)
 		}
 
+		if err := saveMetadata(clusterID, cfg); err != nil {
+			return err
+		}
+
+		go func() {
```
Comment (viktorasm) [inline main.go:179]: temporary hack to test OOM behavior (to be deleted before merging the PR)

File: internal/services/monitor/monitor.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+		},
+	})
+	if err != nil {
+		m.log.Errorf("failed fetching k8s events after agent restart: %v", err)
+	} else {
+		if len(events.Items) == 0 {
+			m.log.Warnf("no k8s events detected for %s/%s", m.pod.Namespace, m.pod.Name)
+		}
+		for _, e := range events.Items {
+			m.log.Errorf("k8s events detected: TYPE:%s REASON:%s TIMESTAMP:%s MESSAGE:%s", e.Type, e.Reason, e.LastTimestamp.UTC().Format(time.RFC3339), e.Message)
```
Comment (viktorasm) [inline internal/services/monitor/monitor.go:73]: TODO: exclude `type:normal` events

File: internal/services/monitor/metadata.go [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+	"strings"
+
+	"github.com/fsnotify/fsnotify"
+	"github.com/sirupsen/logrus"
+)
+
+type Metadata struct {
+	ClusterID string
+	ProcessID uint64
+}
```
Comment (saumas) [inline internal/services/monitor/metadata.go:18]: This struct is marhaled to json, it should have `json` tags.

**Review by saumas — CHANGES_REQUESTED** [review]
Looks good. Small comments.

**Review by saumas — APPROVED** [review]

**Pull request merged**